hail/python/hail/experimental/context.py

```diff
@@ -0,0 +1,147 @@
+from typing import Dict, List, Optional
+
+from py4j.java_gateway import JavaGateway, JavaObject
+from py4j.protocol import Py4JJavaError
+
+from hail.backend.backend import fatal_error_from_java_error_triplet
+from hail.backend.local_backend import LocalBackend
+from hail.backend.py4j_backend import raise_when_mismatched_hail_versions, start_py4j_gateway
+from hail.context import HailContext, _get_log
+from hail.context import init as init_
+from hail.utils.java import BackendType, array_of, choose_backend, scala_object, scala_package_object
+from hail.version import __version__
+from hailtop.aiocloud.aiogoogle import GCSRequesterPaysConfiguration
+from hailtop.config import ConfigVariable, configuration_of, get_remote_tmpdir
+
+
+def init(
+    app_name=None,
+    log=None,
+    quiet=False,
+    append=False,
+    branching_factor=50,
+    tmp_dir=None,
+    default_reference='GRCh37',
+    global_seed=None,
+    _optimizer_iterations=None,
+    *,
+    backend: Optional[BackendType] = None,
+    worker_cores=None,
+    worker_memory=None,
+    batch_id=None,
+    gcs_requester_pays_configuration: Optional[GCSRequesterPaysConfiguration] = None,
+    regions: Optional[List[str]] = None,
+    gcs_bucket_allow_list: Optional[Dict[str, List[str]]] = None,
+    jvm_heap_size: str | None = None,
```

Comment:
Use the 'new' typing syntax.
```suggestion
    backend: BackendType | None = None,
    worker_cores=None,
    worker_memory=None,
    batch_id=None,
    gcs_requester_pays_configuration: GCSRequesterPaysConfiguration | None = None,
    regions: list[str] | None = None,
    gcs_bucket_allow_list: dict[str, list[str]] | None = None,
    jvm_heap_size: str | None = None,
```